### PR TITLE
[code] fix: broken testing due to wrong config setup

### PR DESCRIPTION
**Pull request by grejdi-mbta**
This PR fixes config issues with dev/test database configurations and prod. It also refactors some tests to conform with how tests are setup in general.

### CODE REVIEW

File: ex_cubic_ods_ingestion/test/ex_cubic_ods_ingestion/process_ingestion_test.exs [hunk 1]
```diff
@@ -10,16 +10,25 @@ defmodule ExCubicOdsIngestion.ProcessIngestionTest do
...
+  setup do
+    # insert table
+    table = Repo.insert!(MockExAws.Data.table())
 
+    # insert load records
     {:ok, load_recs} =
       CubicOdsLoad.insert_new_from_objects_with_table(
-        MockExAws.Data.load_objects(),
-        new_table_rec
+        MockExAws.Data.load_objects_without_bucket_prefix(),
```
Comment (grejdi-mbta) [inline ex_cubic_ods_ingestion/test/ex_cubic_ods_ingestion/process_ingestion_test.exs:20]: Wasn't failing, but I made this adjustment to be in alignment with other tests.

File: ex_cubic_ods_ingestion/test/ex_cubic_ods_ingestion/process_ingestion_test.exs [hunk 2]
```diff
@@ -10,16 +10,25 @@ defmodule ExCubicOdsIngestion.ProcessIngestionTest do
   require MockExAws.Data
   require Logger
 
-  defp new_load_recs(_tags) do
-    new_table_rec = Repo.insert!(MockExAws.Data.table())
+  setup do
```
Comment (grejdi-mbta) [inline ex_cubic_ods_ingestion/test/ex_cubic_ods_ingestion/process_ingestion_test.exs:13]: Using setup here like other test files.

File: ex_cubic_ods_ingestion/test/ex_cubic_ods_ingestion/workers/ingest_test.exs [hunk 3]
```diff
@@ -16,7 +16,7 @@ defmodule ExCubicOdsIngestion.Workers.IngestTest do
       # insert load records
       {:ok, new_load_recs} =
         CubicOdsLoad.insert_new_from_objects_with_table(
-          MockExAws.Data.load_objects(),
+          MockExAws.Data.load_objects_without_bucket_prefix(),
```
Comment (grejdi-mbta) [inline ex_cubic_ods_ingestion/test/ex_cubic_ods_ingestion/workers/ingest_test.exs:19]: Was failing if prefixes were set. Now it doesn't.

File: ex_cubic_ods_ingestion/config/runtime.exs [hunk 4]
```diff
@@ -1,24 +1,22 @@
...
 
-config :ex_cubic_ods_ingestion, ExCubicOdsIngestion.Repo,
-  username: System.get_env("DB_USER"),
-  database: System.get_env("DB_NAME"),
-  hostname: System.get_env("DB_HOST"),
-  password: System.get_env("DB_PASSWORD"),
-  port: "DB_PORT" |> System.get_env("5432") |> String.to_integer(),
-  configure: {ExCubicOdsIngestion.Repo, :before_connect, []}
+# overwrite these on prod at runtime, so we can get the container env variables
+if config_env() == :prod do
```
Comment (grejdi-mbta) [inline ex_cubic_ods_ingestion/config/runtime.exs:4]: Took this from [hexpm](https://github.com/hexpm/hexpm/blob/main/config/runtime.exs).

**Comment by paulswartz** [comment]
> This PR fixes config issues with dev/test database configurations and prod. It also refactors some tests to conform with how tests are setup in general.

Can you say more about what the issues were? We had an earlier PR (https://github.com/mbta/data_platform/pull/25) which made changes so that the tests would pass when the prefixes were configured in the test environment.

**Comment by grejdi-mbta** [comment]
> Can you say more about what the issues were? We had an earlier PR (#25) which made changes so that the tests would pass when the prefixes were configured in the test environment, and the tests are passing on the default branch.

This issue is related to database configurations, whereas the PR you are referencing was dealing with prefixes. I had this issue at that time as well, I just needed a little more time to research the various ways configs were setup in other projects. 

Because DB configs were in `runtime.exs`, I was unable to overwrite them for testing, specifically overwriting the database name.

File: ex_cubic_ods_ingestion/config/runtime.exs (hunk 4 above)
Comment (paulswartz) [reply ex_cubic_ods_ingestion/config/runtime.exs:4]: Can you say `config_env() != :test` and reduce the duplication with dev? Or alternatively:
```
...normal configuration...
if config_env() == :test do
  config :ex_cubic_ods_ingestion, ExCubicOdsIngestion.Repo, database: #{System.get_env("DB_NAME")}_test`
end
```

Or have a separate environment file which has the configuration for your test DB? (this last one is what I do)

Comment (grejdi-mbta) [reply ex_cubic_ods_ingestion/config/runtime.exs:4]: How do you load you environment variables? Just curious. I'm inclined to go with alternative approach you provided here, but I know there are various ways of managing environment variables.

Comment (paulswartz) [reply ex_cubic_ods_ingestion/config/runtime.exs:4]:
```
$ env (cat .env) mix run --no-halt
$ env (cat .env.test) mix test
```

Comment (grejdi-mbta) [reply ex_cubic_ods_ingestion/config/runtime.exs]: Thanks. For ease of use I opted to use your second suggestion.

**Review by paulswartz — APPROVED** [review]
🍰

**Pull request merged**